crates/consensus/beacon/src/engine/hooks/controller.rs

```diff
@@ -138,6 +138,12 @@ impl EngineHooksController {
             return Poll::Pending
         }
 
+        // If we don't have a finalized block, we might be on an optimistic sync scenario where we
+        // cannot skip the FCU with the finalized hash, otherwise CL might misbehave.
+        if hook.db_access_level().is_read_write() && args.finalized_block_number.is_none() {
+            return Poll::Pending
+        }
+
```

Comment:
@shekhirin 

ok?

Reply:
that seems fine, I'd just clarify the comment and mention that we skip FCUs when the hook is active